src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,80 @@
+import classNames from 'classnames';
+import React from 'react';
+import { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { FILTER } from '../../App';
+
+interface Props {
+  todos: Todo[];
+  onFilter: (value: string) => void;
+  onDelete: (todoList: Todo[]) => void;
+}
+
+export const Footer: React.FC<Props> = ({ onFilter, onDelete, todos }) => {
+  const [activeLink, setActiveLink] = useState(FILTER.All);
+
+  const handleFiltering = (filterType: string) => {
+    onFilter(filterType);
+    setActiveLink(filterType);
+  };
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {activeTodos.length} items left
+      </span>
+
+      {/* Active link should have the 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          onClick={() => handleFiltering(FILTER.All)}
+          className={classNames(
+            'filter__link',
+            activeLink === FILTER.All && 'selected',
+          )}
+          data-cy="FilterLinkAll"
+        >
+          All
+        </a>
+
+        <a
+          href="#/active"
+          onClick={() => handleFiltering(FILTER.ACTIVE)}
+          className={classNames(
+            'filter__link',
+            activeLink === FILTER.ACTIVE && 'selected',
+          )}
+          data-cy="FilterLinkActive"
+        >
+          Active
+        </a>
+
+        <a
+          href="#/completed"
+          onClick={() => handleFiltering(FILTER.COMPLETED)}
+          className={classNames(
+            'filter__link',
+            activeLink === FILTER.COMPLETED && 'selected',
```

Comment:
Use the classnames library rules for add classes with condition, fix it everywhere